src/main/java/edu/illinois/cs/dt/tools/utility/iFixPlusPlugin.java

```diff
@@ -0,0 +1,744 @@
+        dtname= Configuration.config().getProperty("replay.dtname");
+        output= Configuration.config().getProperty("replay.output");
+        slug= Configuration.config().getProperty("replay.slug");
+        subxmlFold = Configuration.config().getProperty("replay.subxmlFold");
+        rootFold = Configuration.config().getProperty("replay.rootFold");
+        tmpfile = Configuration.config().getProperty("replay.tmpfile");
+        module = Configuration.config().getProperty("replay.module");
+        diffFieldsFold = Configuration.config().getProperty("replay.diffFieldsFold");
+        subdiffsFold = Configuration.config().getProperty("replay.subdiffsFold");
+        reflectionFold = Configuration.config().getProperty("replay.reflectionFold");
+
+        int xmlFileNum = countDirNums(subxmlFold);
+        System.out.println("xmlFileName: " + xmlFileNum);
+
+        if (runner != null && module.equals(PathManager.modulePath().toString())) {
+            System.out.println("replyPath: " + replayPath);
+            System.out.println("module: " + module);
+
+            try {
+                Files.write(Paths.get(output),
+                        (lastPolluter() + ",").getBytes(),
+                        StandardOpenOption.APPEND);
+                // phase 0 check json file
+                System.out.println("~~~~~~ Begin to check if this test is a true order-dependent test.");
+                Configuration.config().properties().setProperty("statecapture.phase", "check");
+                if (testFailOrder()==null) {
+                    timing(startTime);
+                    Files.write(Paths.get(output),
+                            "0,0,0,0,0,0,0,wrongjsonfail,".getBytes(),
+                            StandardOpenOption.APPEND);
+                    System.out.println("## Original json file is wrong!");
+                    return;
+                }
+
+                if (testPassOrder_full()==null) {
+                    timing(startTime);
+                    Files.write(Paths.get(output),
+                            "0,0,0,0,0,0,0,wrongjsonpass,".getBytes(),
+                            StandardOpenOption.APPEND);
+                    System.out.println("## Original json file is wrong!");
+                    return;
+                }
+
+                for(int i=0; i<10; i++) {
+                    Try<TestRunResult> phase0ResultFail = null;
+                    try {
+                        phase0ResultFail = runner.runList(testFailOrder());
+                    }
+                    catch (Exception ex) {
+                        System.out.println("## Encountering error when checking the failing order: " + ex);
+                    }
+
+                    System.out.println("Failing order results: " +
+                            phase0ResultFail.get().results().get(dtname).result().toString());
+
+                    if (phase0ResultFail.get().results().get(dtname).result().toString().equals("PASS")) {
```

Comment:
Since the variable gets initalized to `null` and can potentially not be set properly in the try block, this could actually lead to a `NullPointerException`. That kind of defeats the purpose of using the `Try` construct, since the point of that is to not have to check for null (you would first check whether it has a value).